week01/four_sum.cpp

```diff
@@ -0,0 +1,48 @@
+//TC: O(N^3)
+//SC: O(N)
+
+class Solution {
+public:
+    vector<vector<int>> fourSum(vector<int>& nums, int target) {
+        sort(nums.begin(), nums.end()); /// sort for apply two sum
+        vector<vector<int>>result; 
+        int len = nums.size();
+
+        for(int i = 0; i < len - 3; i++)
```

Comment:
```suggestion
        for(int i = 0; i < len; i++)
```

Please don't use this kind of hacks. This kind of optimizations are not actually optimizations. They make the whole code buggy/less readable. We call it `magic numbers`.